src/Components/Server/src/ComponentHub.cs

```diff
@@ -66,7 +66,15 @@ public override Task OnDisconnectedAsync(Exception exception)
             }
 
             CircuitHost = null;
-            return _circuitRegistry.DisconnectAsync(circuitHost, Context.ConnectionId);
+            if (exception != null)
```

Comment:
This definitely needs some comments explaining why this matters and what kinds of scenarios it supports.

Reply:
Will add some comments.

Reply:
I added some comments, but on further thoughts I'm not sure this will do what we want when used in combination with the signalr service. The reason for it is that as far as I know the connection to the server will be kept alive in the signalr service case and only the connection between the client and the server will be closed (hence you won't get a nice disconnectedasync notification.

Am I wrong? @anurse @BrennanConroy

Reply:
Where are you missing the `OnDisconnectedAsync` call? It's my understanding you'll definitely get one if the WebSocket closes gracefully even when using the service. I'm not clear what the concern is.

Reply:
> as I know the connection to the server will be kept alive in the signalr service case

Do you mean the connection from the App Server to the Service? Yes, that lives for the duration of the app. The virtual connection from Client to App Server (via the Service) is disconnected though and you'll be notified of that here.

Reply:
Then that's great!